src/modeler/AutoRest.Swagger/Validation/TrackedResourceListByImmediateParent.cs

```diff
@@ -35,6 +35,8 @@ class TrackedResourceListByImmediateParent : TypedRule<Dictionary<string, Dictio
         /// </summary>
         public override Category Severity => Category.Warning;
 
+        private Regex pathRegEx = new Regex("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/[^/]+/[^/]+/{[^/]+}.*/(\\w+)$", RegexOptions.IgnoreCase);
```

Comment:
can you add a comment on what you're trying to find with regex? It'd be good for someone that reads the code to get an idea of what you're looking to do without having to understand the whole regex, particularly since we're not using a shared one for this case.